### PR TITLE
Admin tracing

### PR DESCRIPTION
**Pull request by rivera-fernando**

### CODE REVIEW

**Review by christinachance — APPROVED** [review]

File: project/src/main/java/com/google/sps/servlets/Exposure.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        personAndPlace.add((String) entity.getProperty("lat"));
+        personAndPlace.add((String) entity.getProperty("lng"));
+
+
+        data.add(personAndPlace);
+      }
+      response.setContentType("application/json;");
+      response.getWriter().println(gson.toJson(data));
+  }
+}
```
Comment (christinachance) [inline project/src/main/java/com/google/sps/servlets/Exposure.java:124]: add new line

Comment (rivera-fernando) [reply project/src/main/java/com/google/sps/servlets/Exposure.java:124]: done

**Review by Tcintra — APPROVED** [review]

File: project/src/main/java/com/google/sps/servlets/Exposure.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+              datastore.put(entity);
+              session.setAttribute("diagnosed", "negative");
+          }
+          return;
+      }
+      for (Entity entity: results.asIterable()) {
+          entity.setProperty("diagnosed", "positive");
+          datastore.put(entity);
+          session.setAttribute("diagnosed", "positive");
+      }
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/Exposure.java:74]: Maybe 66-78 could be something like below. more cleaner:
```
diagnosed = action.equals("revert")? 'negative' : 'positive'
for (Entity entity: results.asIterable()) {
          entity.setProperty("diagnosed", diagnosed);
          datastore.put(entity);
          session.setAttribute("diagnosed", diagnosed);
}
```

File: project/src/main/java/com/google/sps/servlets/Exposure.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+      
+      query = new Query("LocationLog").setFilter(emailFilter);
+      results = datastore.prepare(query);
+      PreparedQuery all;
+      Map<String, String> emails = new HashMap<String, String>();
+      for (Entity entity : results.asIterable()) {
+          Filter dateFilter = 
+            new FilterPredicate("date", FilterOperator.EQUAL, entity.getProperty("date"));
+          Filter latFilter = 
+            new FilterPredicate("lat", FilterOperator.EQUAL, entity.getProperty("lat"));
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/Exposure.java]: `lat` could you please use full words , same as `lng`

**Review by wulian1223 — APPROVED** [review]
LGTM after fixing comments.

File: project/src/main/java/com/google/sps/servlets/Exposure.java (hunk 2 above)
Comment (rivera-fernando) [reply project/src/main/java/com/google/sps/servlets/Exposure.java:74]: Done.

File: project/src/main/java/com/google/sps/servlets/Exposure.java (hunk 3 above)
Comment (rivera-fernando) [reply project/src/main/java/com/google/sps/servlets/Exposure.java]: Done.

**Pull request merged**